dissect/target/plugins/os/windows/firewall.py

```diff
@@ -0,0 +1,329 @@
+                if protocol := data.get("protocol"):
+                    if protocol.isnumeric():
+                        data["protocol"] = IANAProtocol(int(protocol)).name
+                    elif protocol.isalpha():
+                        data["protocol"] = protocol.upper()
+
+                # Make sure actions and directions are consistent
+                if "action" in data:
+                    data["action"] = data["action"].upper()
+                if "dir" in data:
+                    data["dir"] = data["dir"].upper()
+
+                # Put port ranges in a list
+                if "lport" in data:
+                    data["lport"] = data["lport"].split(",")
+                if "rport" in data:
+                    data["rport"] = data["rport"].split(",")
+
+                r.append(("path", "source"))
+
+                yield TargetRecordDescriptor("windows/firewall/rule", r)(
+                    key=entry.name,
+                    **data,
+                    source=f"HKLM\\{reg.path}",
+                    _target=self.target,
+                )
+
+    @export(record=WindowsFirewallLogRecord)
+    def logs(self) -> Iterator[WindowsFirewallLogRecord]:
+        """Parse Windows Firewall log files.
+
+        Currently parses ``pfirewall*`` files in ``sysvol\\Windows\\System32\\LogFiles\\Firewall\\`` only.
+        Does not yet parse dynamically set log locations e.g. ``netsh advfirewall set currentprofile logging filename``.
+
+        Resources:
+            - https://learn.microsoft.com/en-us/windows/security/operating-system-security/network-security/windows-firewall/configure-logging
+
+        Yields Windows Firewall log records with the following fields:
+
+        .. code-block:: text
+
+            ts (datetime): The timestamp of the log entry.
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            action (string): Allow or Block.
+            protocol (string): TCP, UDP or other IANA protocol value.
+            src_ip (net.ipaddress): Source IP address.
+            dst_ip (net.ipaddress): Destination IP address.
+            src_port (varint): Source port number.
+            dst_port (varint): Destination port number.
+            size (filesize): Size in bytes of the packet(s).
+            tcpflags (string): TCP header control flags.
+            tcpsyn (string): TCP sequence number.
+            tcpack (string): TCP acknowledgement number.
+            tcpwin (string): TCP window size in bytes.
+            icmptype (string): ICMP packet type.
+            icmpcode (string): ICMP packet code.
+            info (string): Additional information.
+            path (string): Direction of the traffic, either SEND, RECEIVE, FORWARD or UNKNOWN.
+            source (path): Source path of the record log line.
+        """
+
+        for path in self.log_paths:
+            try:
+                fh = path.open("rt")
```

Comment:
Can you use a `with` statement? I'm trying to get new code to be a little bit more aware of that.